### ISSUE TITLE
`edgedb instance link` allow for invalid instance name?

### ISSUE THREAD

**Issue by mhammerc** [edgedb/edgedb-cli#571]
Hello!!

- EdgeDB Version: rc2
- EdgeDB CLI Version: rc2
- OS Version: MacOS 12

Steps to Reproduce:

1. Run a standalone `edgedb-server` instance.
2. Run `edgedb instance link --dsn edgedb://admin@127.0.0.1:5656/edgedb` which gives the following output:
```
<... regular questions, nothing special>

Specify a new instance name for the remote server [default: 127_0_0_1]:
> 127_0_0_1

Successfully linked to remote instance. To connect run:
  edgedb -I 127_0_0_1
```

3. Instance name registered to CLI is `127_0_0_1`.
4. Run `edgedb instance list`

```
┌───────────┬──────┬──────────┬──────────────┬─────────┐
│ Name      │ Port │ Version  │ Installation │ Status  │
├───────────┼──────┼──────────┼──────────────┼─────────┤
│ 127_0_0_1 │ 5656 │ 1.0-rc.2 │ remote       │ running │
└───────────┴──────┴──────────┴──────────────┴─────────┘
```

5. Run `edgedb -I 127_0_0_1` as shown above
`edgedb error: ClientError: instance name "127_0_0_1" contains unsupported characters`

6. Run `edgedb instance unlink 127_0_0_1` (to start over with a fresh name 😂)

```
error: Invalid value for '<name>': instance name must be a valid identifier, (regex: ^[a-zA-Z_][a-zA-Z_0-9]*$)

For more information try --help
```

**TLDR**: From the CLI, linking a remote instance started locally using 127.0.0.1 gives default name of `127_0_0_1` from the interactive prompt. This name doesn't match the required regex `^[a-zA-Z_][a-zA-Z_0-9]*$` (according to `edgedb instance unlink`)

https://regex101.com/r/3tMaEK/1

**Comment by mhammerc** [comment]
good job, thanks! 👍
